### PR TITLE
refactor(formats): consolidate the dataframe conversion logic

### PR DESCRIPTION
**Pull request by kszucs**
Squash the multipledispatch `sch.convert()` overloads in the pandas and dask format implementations.
Turned out that we don't need the dask overloads, we just carefully need to inject a `.compute()` call to the pandas series inference where we have object dtypes, so the dask format is not necessary anymore. Also deprecated the `Schema.apply_to()` method. 

After taking a quick look at the new `conver_pandas_series()` function's coverage we can possibly simplify it, but I didn't dig deeper.

depends on https://github.com/ibis-project/ibis/pull/6294

### CODE REVIEW

**Comment by kszucs** [comment]
@cpcloud should be ready for review

File: ibis/formats/pandas.py [hunk 1]
```diff
@@ -64,7 +64,9 @@ def schema_from_pandas(schema):
     return sch.schema(ibis_types)
 
 
-def schema_from_pandas_dataframe(df: pd.DataFrame, schema=None):
+def schema_from_pandas_dataframe(
+    df: pd.DataFrame, schema=None, inference_function=infer_sequence_dtype
```
Comment (cpcloud) [inline ibis/formats/pandas.py:68]: Do we need to expose the optionality of `inference_function` here?

File: ibis/formats/pandas.py [hunk 2]
```diff
@@ -87,109 +89,101 @@ def schema_from_pandas_dataframe(df: pd.DataFrame, schema=None):
...
-        return column.astype(out_dtype.to_pandas())
-    except pd.errors.OutOfBoundsDatetime:
-        try:
-            return column.map(date_parse)
-        except TypeError:
-            return column
-    except TypeError:
-        column = pd.to_datetime(column)
-        timezone = out_dtype.timezone
+    if dtype.is_boolean():
```
[the remark this answers is not in the document]
Comment (kszucs) [reply ibis/formats/pandas.py:121]: I think we will need new code paths for that.

File: ibis/formats/pandas.py (hunk 1 above)
Comment (kszucs) [reply ibis/formats/pandas.py:68]: You mean `_inference_function`?

File: ibis/formats/pandas.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (cpcloud) [reply ibis/formats/pandas.py:121]: Ok, seems fine!

**Review by cpcloud — APPROVED** [review]

**Pull request merged**